### PR TITLE
bring back provider headers and checksum from older oceanJs function

### PR DESCRIPTION
**Pull request by soonhuat**
Fixes # .

Changes proposed in this PR:

- bring back headers for provider getFileInfo
- bring back checksum flag for provider getFileInfo
- add back withChecksum signature comment

### CODE REVIEW

File: src/services/Provider.ts [hunk 1]
```diff
@@ -192,20 +193,24 @@ export class Provider {
...
+   * @param {any} headers headers key value pairs associated with the asset GET request
    * @return {Promise<FileInfo[]>} urlDetails
    */
   public async getFileInfo(
     file: UrlFile | Arweave,
     providerUri: string,
-    signal?: AbortSignal
+    withChecksum: boolean = false,
+    signal?: AbortSignal,
+    headers?: any
```
Comment (bogdanfazakas) [inline src/services/Provider.ts:206]: can't we define a interface for this to avoid using `any` , with the desired HTTP get fields needed ?

**Comment by bogdanfazakas** [comment]
it would be also nice to have a test in the `test/integration/Provider.test.ts` file that covers an example of `getFileInfo` using headers

**Comment by alexcos20** [comment]
We should add headers to UrlFile and Graphql types

**Comment by bogdanfazakas** [comment]
handled in #1665

**Pull request closed without merging**